### ISSUE TITLE
Error in generating Thermal ACE file

### ISSUE THREAD

**Issue by Mahdi-20** [njoy/NJOY2016#219]
Dear experts,
I am new to NJOY. I just tried to generate the thermal cross-sections for H in CH2 from ENDF library using NJOY21.
But I faced the following error. I searched the issues but I could find any similar error. 
I have attached my input files together with the ENDF files. Any help is really appreciated.

[error] Encountered invalid value for iza
[info] Error while parsing line 34

1001/
^

[info]
The iza argument specifies the ZA values for the moderator components.

[info] Trouble while validating card 8a
[info] Trouble validating ACER input.
[info] Error while running NJOY21

[input.txt](https://github.com/njoy/NJOY2016/files/7120768/input.txt)
[tape20.txt](https://github.com/njoy/NJOY2016/files/7120769/tape20.txt)
[tape21.txt](https://github.com/njoy/NJOY2016/files/7120770/tape21.txt)

**Comment by paulromano** [comment]
@Mahdi-20 NJOY2016 and NJOY21 now take a fourth entry on ACER card 8 that indicates the number of moderator component ZA values. If you change your card 8 to `125 350.0 'poly' 1 /`, NJOY21 should then accept the next line with only one ZA value. The NJOY devs will have to chime in on whether the current behavior is a bug for NJOY21, given that NJOY2016 accepts your input as is with no problem.

**Comment by Mahdi-20** [comment]
@paulromano Thank you so much it worked!
But, there are questions that I cannot figure out. 
1) I have generated the ACE format of H in CH2 at 350K using the tsl data [taken from ENDF/B-VII] which is named "tape28". 
  Then I compared it with the original ACE format that is already generated by ENDF/B-VII named "h-poly".
  As it is seen from the screenshots below, there are some differences between the 2 ACE files such as the number of lines, 
  number of reactions, etc. However, I have used the same ENDF format file. 

![h-poly](https://user-images.githubusercontent.com/90252318/132428013-ec20cd23-25c4-4b3c-b6bc-a464f8882fdd.png)
![tape28](https://user-images.githubusercontent.com/90252318/132428015-1fef1dcc-6715-49a9-ab18-f44e66d60f27.png)


2) I want to generate H in CH2 at 77 K, where can I find the necessary parameters for the "leaper" module such as rho(e), alpha, Beta, etc. for temperature 77K?

**Comment by paulromano** [comment]
There could be several reasons for the differences: different versions of NJOY used or different parameter choices from the NJOY runs are two possibilities.

The ENDF/B-VII.1 evaluation for H in CH2 doesn't have data at 77 K but the more recent ENDF/B-VIII.0 evaluation does. The [ENDF/B-VIII.0 distribution](https://www.nndc.bnl.gov/endf/b8.0/zips/ENDF-B-VIII.0_thermal_scatt.zip) also includes LEAPR inputs.

**Comment by marquezj** [comment]
@Mahdi-20 you have different reconstruction parameters in the two files.

- The first one has 22 angular bins ([nbin in THERMR](https://github.com/njoy/NJOY2016/blob/6ef2a1d04ba3cad3c114abaaf6069e317364158d/src/thermr.f90#L107)), 80 outgoing energy bins ([nbint in ACER](https://github.com/njoy/NJOY2016/blob/6ef2a1d04ba3cad3c114abaaf6069e317364158d/src/acer.f90#L183)) and a continuous energy distribution ([iwt in ACER](https://github.com/njoy/NJOY2016/blob/6ef2a1d04ba3cad3c114abaaf6069e317364158d/src/acer.f90#L191)).
- The second one has 16 angular bins, 64 outgoing energy bins and a discrete energy distribution.

Also there is a different number of incoming energies, which might come from using different versions of NJOY or a higher maximum energy (emax in [ACER](https://github.com/njoy/NJOY2016/blob/6ef2a1d04ba3cad3c114abaaf6069e317364158d/src/acer.f90#L189) and [THERMR](https://github.com/njoy/NJOY2016/blob/6ef2a1d04ba3cad3c114abaaf6069e317364158d/src/thermr.f90#L135)).

If you are using a modern Monte Carlo code (MCNP6 / OpenMC) you can use the continuous energy distribution, which is better (although it creates larger files). If you use an older code (MCNP5 / MCNPX / PHITS) you need to use the older discrete format.

**Comment by Mahdi-20** [comment]
@paulromano Thank you for your response. I have checked the LEAPER inputs of ENDF/B-VIII.0. 
For my project, I need the ACE format cross-section data of H in CH2 in lower temperatures as well, say, 40K. 
As I have studied this issue, the alpha and beta grids are the same for each temperature, if I understood well.
However, for the LEAPER module, I need the photon frequency distributions at 40K. I am wondering how these frequency distributions are calculated for each tempreture. 
Thanks for your time.

**Comment by paulromano** [comment]
The phonon frequency distributions can be inferred from experimental measurements or calculated via molecular dynamics simulations. This is really in the realm of data evaluators (like @marquezj) and would not be something that a regular user would  typically do.

**Comment by marquezj** [comment]
@Mahdi-20 as you can see in the H-CH2 input file, the phonon spectrum is considered to be constant with temperature. Actually, for solids, they are usually calculated at 0 K with DFT (although in this particular case it was calculated with MD, but that is a separate discussion on the pros and cons of each approach). If you just need a quick test, use the existing phonon spectrum and grid (scaled with lat=1) and change the temperature.

In any case, polyethylene is not usually used below liquid nitrogen temperature because because the cryogenic cost is not worth it. You should look at solid methane (or solids with methyl groups) or liquid hydrogen and deuterium.

**Comment by Mahdi-20** [comment]
Dear experts,
Regarding my last questions, I tried to use some NJOY-processed cross-section data to consider the effect of temperature on the neutron flux.
I used the tsl-ACE data from ENDF/B-VIII.0 for H in Polyethelene(PE) at two temperatures of 296K and 77 K. 
I used the PHITS code to calculate the neutron flux after a slab of PE at the mentioned temperatures. However, the results in both cases are almost similar and I expected to see the Maxwellian peak shifted down in the energy (as shown in the Fig).
It seems that the tsl data at each temperature are almost similar, is my understanding correct?

Thanks for any comments.

![image](https://user-images.githubusercontent.com/90252318/137883986-9afc940e-a7b5-4581-87dd-5377b543f037.png)
